src/components/Card.tsx

```diff
@@ -45,18 +49,42 @@ export const Card = (item: History): JSX.Element => {
         <p className={cardClassnames.data}>Murojaah by using {item.murojaahMethod}</p>
         <p className={cardClassnames.data}>Total Murojaah is {item.totalMurojaah}</p>
         <p className={cardClassnames.date}>{item.occuredAt}</p>
+      </>
+    );
+  };
+
+  const BaseCard = (children: JSX.Element): JSX.Element => {
+    const btnClass: Record<string, string> = {
+      base: 'p-2 rounded',
+      edit: 'bg-yellow-500 hover:bg-yellow-700',
+      delete: 'bg-red-500 hover:bg-red-700',
+    };
+
+    const [isButtonsVisible, setIsButtonsVisible] = useState(false);
+
+    function toggleButtons(): void {
+      isButtonsVisible ? setIsButtonsVisible(false) : setIsButtonsVisible(true);
+    }
+
+    return (
+      <div className={cardClassnames.container}>
+        <div onClick={toggleButtons}>{children}</div>
+        {isButtonsVisible && (
+          <div className="flex flex-col gap-2 w-full mt-2">
+            <button className={clsx(btnClass.base, btnClass.edit)}>Edit</button>
+            <button className={clsx(btnClass.base, btnClass.delete)}>Delete</button>
+          </div>
+        )}
       </div>
     );
   };
 
   switch (item.murojaahType) {
     case MurojaahType.Juz:
-      return JuzCard();
+      return BaseCard(JuzCard());
     case MurojaahType.Surah:
-      return SurahCard();
+      return BaseCard(SurahCard());
     case MurojaahType.Ayah:
-      return AyahCard();
-    default:
-      return <></>;
+      return BaseCard(AyahCard());
   }
```

Comment:
can this be wrapped in useMemo(), looks like quite complex

Reply:
is this correct https://github.com/RR-Ardhika/murojaah/pull/9/commits/85ee144e1cd61dbd68d7ea534fcf6532ccffe372

also got this warning
![image](https://github.com/RR-Ardhika/murojaah/assets/14792658/22e44553-c99c-41b4-976b-039d0eed87dd)

Reply:
please include the missing dependencies if it works

Reply:
for fetchData it didn't work, got stuck when choosing juz in juz dropdown

Reply:
for the cardClassnames solved in https://github.com/RR-Ardhika/murojaah/pull/9/commits/6b48662e81200eacafdfe4be36e43276682066fc, thank you

Reply:
already wrapper in https://github.com/RR-Ardhika/murojaah/pull/9/commits/5a4041a4b899b581db984b8195c1bdd952e8a53d but got error

![image](https://github.com/RR-Ardhika/murojaah/assets/14792658/3600ecf8-9f29-47e6-a4d3-cffa3e3badf4)